src/cli.rs

```diff
@@ -110,3 +165,80 @@ impl TryFrom<&'_ clap::ArgMatches<'static>> for Command {
     }
 }
 
+
+impl TryFrom<&'_ clap::ArgMatches<'static>> for WgCommand {
+    type Error = Error;
+    fn try_from(args: &clap::ArgMatches<'static>) -> Result<Self> {
+
+
+            if let Some(wg_command) = args.subcommand_matches("wireguard")
+            {
+                // Unlock
+                if let Some (command) = wg_command.subcommand_matches("unlock")
+                {
+                        let pin = match command.value_of("PIN") {
+                            Some(s) => {s},
+                            None => {return Err(anyhow::anyhow!("Could not parse pin"));}
+    
+                        };
+                       return Ok(WgCommand::Unlock(Unlock {
+                        pin : pin.to_string()
+                    })
+                )}
+            
+                // AEAD
+                if let Some (command) = wg_command.subcommand_matches("aead")
+                {
+                    let c = match command.value_of("C") {
+                        Some(s) => {hex_string_to32_char_array(s)},
+                        None => {return Err(anyhow::anyhow!("Could not parse chained key value C"));}
+                    };
+
+                    let h = match command.value_of("H") {
+                        Some(s) => {hex_string_to32_char_array(s)},
+                        None => {return Err(anyhow::anyhow!("Could not parse chained hash value H"));}
+                    };
+
+                    let pubkey = match command.value_of("PUBKEY") {
+                        Some(s) => {hex_string_to32_char_array(s)},
+                        None => {return Err(anyhow::anyhow!("Could not parse public key"));}
+                    };
+
+                    let key_id = match command.value_of("I") {
+                        Some(s) => {s.parse()},
+                        None => {return Err(anyhow::anyhow!("Could not parse key id"));}
+                    };
+
+
+                       return Ok(WgCommand::GetAead(GetAead {
+                        pubkey : pubkey.unwrap(),
+                        c : c.unwrap(),
+                        h : h.unwrap(),
+                        key_id : key_id.unwrap()
+                    })
+                )}
+
+
+                }
+
+           Err(anyhow::anyhow!("Unexpected case"))
+    }
+}
+
+fn hex_string_to32_char_array( s : &str ) -> Result<[u8; 32]>
+{
+    if s.chars().count() > 32
+    {
+       return Err(anyhow::anyhow!("String too long"));
+    }
+
+    let mut arr : [u8; 32] = [0;32];
+    let chars = hex::decode( s);
+
+    for (index, c) in chars.unwrap().iter().enumerate()
+    {
+        arr[index] = c.clone();
+    }
+
+   return Ok(arr.clone())
+}
```

Comment:
Try getting acquainted with "standard Rust traits", mainly From/Into and their fallible variants TryFrom/TryInto.

Also combinators on Result, such as map/map_err, and the `?` operator.

E.g. here something like:
```
let decoded = hex::decode(s)
use core::conver::TryFrom;
let arr: [u8; 32] = decoded.try_int().map_err(|_| anyhow::anyhow!("String too long"))?;
Ok(arr)
```

You can also always run `cargo clean; cargo clippy` to get hints from the compiler. E.g. bytes are `Copy`, so these clones are redundant.